osc-ui/src/main/java/org/osc/core/broker/view/vc/BaseVCWindow.java

```diff
@@ -303,13 +303,20 @@ protected void handleException(final Exception originalException) {
             exception = originalException.getCause();
 
             // TODO this exception leaks large amounts of implementation detail out of the API
-            if (errorType == ErrorType.PROVIDER_EXCEPTION) {
-                if (RestClientException.isConnectException(exception) && isOpenstack()) {
-                    // Keystone Connect Exception
-                    contentText = VmidcMessages.getString(VmidcMessages_.VC_CONFIRM_IP, KEYSTONE_CAPTION);
-                }
-            } else if (errorType == ErrorType.CONTROLLER_EXCEPTION) {
-                if (RestClientException.isCredentialError(exception)) {
+			if (errorType == ErrorType.PROVIDER_EXCEPTION) {
+				if (RestClientException.isConnectException(exception) && isOpenstack()) {
```

Comment:
since we are checking isOpenstack for all the conditions, can we re-structure this if-else ladder to be more consice and clear?

if(isOpenstack()) {
switchcase or a if-else ladder
}